rllib/core/rl_trainer/tf/tf_rl_trainer.py

```diff
@@ -156,7 +226,8 @@ def add_module(
                 set_optimizer_fn=set_optimizer_fn,
                 optimizer_cls=optimizer_cls,
             )
-        self.traced_update_fn = tf.function(self._do_update_fn)
+        if self._enable_tf_function:
+            self._update_fn = tf.function(self._do_update_fn)
```

Comment:
A note after discussing offline: You need retrace after adding a module, because there is no way for tf tracer to automatically retrace if the logic inside the forward_pass is changed. Therefore, you need to re-assign the update_fn pointer to a re-traced function with the new module added as part of the computational graph.

Reply:
should we add a TODO about this then?
seems important? TODO(kourosh, avnish)?

Reply:
It's already doing this. I just provided more context for myself and the next reviewer (e.g you 😃)

Reply:
oh I see. that makes more sense.